### PR TITLE
273 - use builder pattern in GraphQL top level object

### PR DESCRIPTION
**Pull request by bbakerman**
#273 - to allow consistency with other parts of the API and extensibility in the future, the builder pattern should be used on the GraphQL object

### CODE REVIEW

File: src/main/java/graphql/GraphQL.java [hunk 1]
```diff
@@ -66,7 +119,7 @@ public ExecutionResult execute(String requestString, String operationName, Objec
             RecognitionException recognitionException = (RecognitionException) e.getCause();
             SourceLocation sourceLocation = new SourceLocation(recognitionException.getOffendingToken().getLine(), recognitionException.getOffendingToken().getCharPositionInLine());
             InvalidSyntaxError invalidSyntaxError = new InvalidSyntaxError(sourceLocation);
-            return new ExecutionResultImpl(Arrays.asList(invalidSyntaxError));
+            return new ExecutionResultImpl(Collections.singletonList(invalidSyntaxError));
```
Comment (bbakerman) [inline src/main/java/graphql/GraphQL.java]: Smidge more efficient and IDEA inspections (which I trust mostly) says to do it. 

Broken window theory in code and all...

File: README.md [hunk 2]
```diff
@@ -94,7 +94,9 @@ public class HelloWorld {
         GraphQLSchema schema = GraphQLSchema.newSchema()
                         .query(queryType)
                         .build();
-        Map<String, Object> result = (Map<String, Object>) new GraphQL(schema).execute("{hello}").getData();
+        
+        GraphQL graphQL = GraphQL.newObject(schema).build();
```
Comment (dminkovsky) [inline README.md]: GraphQL.newGraphQL() I think is more in line with the other builder factory names. (newObject() is already "taken" by GraphQLObjectType)

File: src/main/java/graphql/GraphQL.java [hunk 3]
```diff
@@ -29,16 +29,69 @@
 
     private static final Logger log = LoggerFactory.getLogger(GraphQL.class);
```
Comment (dminkovsky) [inline src/main/java/graphql/GraphQL.java]: I think this should take into account the mutation strategy selection, which is already released https://github.com/graphql-java/graphql-java/pull/240/files

Comment (bbakerman) [inline src/main/java/graphql/GraphQL.java]: Yeah I am updating that now

Comment (dminkovsky) [inline src/main/java/graphql/GraphQL.java]: thanks!

File: README.md (hunk 2 above)
Comment (bbakerman) [inline README.md]: Done!

**Pull request merged**